Zend/zend_execute.c

```diff
@@ -830,6 +830,24 @@ ZEND_COLD zend_never_inline void zend_verify_property_type_error(zend_property_i
 	zend_string_release(type_str);
 }
 
+ZEND_COLD zend_never_inline void zend_magic_get_property_type_inconsistency_error(zend_property_info *info, zval *property)
+{
+	zend_string *type_str;
+
+	/* we _may_ land here in case reading already errored and runtime cache thus has not been updated (i.e. it contains a valid but unrelated info) */
+	if (EG(exception)) {
+		return;
+	}
+
+	type_str = zend_type_to_string(info->type);
```

Comment:
```suggestion
	zend_string *type_str = zend_type_to_string(info->type);
```

nit: Personal preference: combining would be clearer. We've been combining declaration and initialization for a while in new code of php-src, but there's a lot of code existing left over which always has declarations at the top for older c compiler versions?

I'd agree that incompatible is consistent with other error messages.

Otherwise it looks fine.